nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -4862,4 +4794,46 @@ mod tests {
             );
         }
     }
+
+    #[tokio::test]
+    async fn insert_target_query_sql_snapshot() {
+        use crate::db::raw_query_builder::expectorate_query_contents;
+
+        let query = InsertTargetQuery {
+            target_id: BlueprintUuid::nil(),
+            enabled: true,
+            time_made_target: Utc::now(),
+        };
+
+        expectorate_query_contents(
+            &query.to_query(),
+            "tests/output/insert_target_blueprint_query.sql",
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn insert_target_query_is_valid() {
+        use crate::db::explain::ExplainableAsync;
+
+        let logctx = dev::test_setup_log("insert_target_query_is_valid");
+        let db = TestDatabase::new_with_pool(&logctx.log).await;
+        let pool = db.pool();
+        let conn = pool.claim().await.unwrap();
+
+        let query = InsertTargetQuery {
+            target_id: BlueprintUuid::nil(),
+            enabled: false,
+            time_made_target: Utc::now(),
+        };
+
+        let _ = query
+            .to_query()
+            .explain_async(&conn)
+            .await
```

Comment:
nit, take it or leave it: i find it useful to also `eprintln!` the `EXPLAIN` output in these, so that you can run `cargo nextest run {name_of_test} --success-output final` if you want to see the query plan.

also, the other `EXPLAIN` tests generally include assertions that there's no `FULL SCAN` in the explanation, is that worth having here?